modules/lib/lib-content/src/main/resources/lib/xp/content.ts

```diff
@@ -11,12 +11,20 @@ declare global {
     interface XpLibraries {
         '/lib/xp/content': typeof import('./content');
     }
-
-    interface XpXData {
-        [key: string]: Record<string, Record<string, unknown>>
-    }
 }
 
+import type {Attachment, Component, Content, PublishInfo, Region} from '@enonic-types/core';
+
+export type {Attachment, PublishInfo, Content, Component, Region} from '@enonic-types/core';
+
+export type Attachments = Content['attachments'];
+
+export type ContentInheritType = Content['inherit'];
+
+export type Workflow = Content['workflow'];
```

Comment:
I would not export these types from the library. I would expect the app-developers to be able to create these types as they need them.

Maybe I would add an example in the documentation on how to do this.